activerecord/test/cases/migration/columns_test.rb

```diff
@@ -346,6 +346,20 @@ def test_change_column_null_with_non_boolean_arguments_raises
         assert_equal "change_column_null expects a boolean value (true for NULL, false for NOT NULL). Got: #{{ from: true, to: false }}", e.message
       end
 
+      def test_change_column_null_does_not_change_default_functions
+        skip unless current_adapter?(:Mysql2Adapter, :TrilogyAdapter) && supports_default_expression?
+
+        function = connection.mariadb? ? "(current_timestamp())" : "(now())"
```

Comment:
It seems it's also `current_timestamp()` for **innodb** in mysql8: https://dev.mysql.com/doc/refman/8.4/en/timestamp-initialization.html
> To specify automatic properties, use the DEFAULT CURRENT_TIMESTAMP and ON UPDATE CURRENT_TIMESTAMP clauses in column definitions. The order of the clauses does not matter. If both are present in a column definition, either can occur first. Any of the synonyms for [CURRENT_TIMESTAMP](https://dev.mysql.com/doc/refman/8.4/en/date-and-time-functions.html#function_current-timestamp) have the same meaning as [CURRENT_TIMESTAMP](https://dev.mysql.com/doc/refman/8.4/en/date-and-time-functions.html#function_current-timestamp). These are [CURRENT_TIMESTAMP()](https://dev.mysql.com/doc/refman/8.4/en/date-and-time-functions.html#function_current-timestamp), [NOW()](https://dev.mysql.com/doc/refman/8.4/en/date-and-time-functions.html#function_now), [LOCALTIME](https://dev.mysql.com/doc/refman/8.4/en/date-and-time-functions.html#function_localtime), [LOCALTIME()](https://dev.mysql.com/doc/refman/8.4/en/date-and-time-functions.html#function_localtime), [LOCALTIMESTAMP](https://dev.mysql.com/doc/refman/8.4/en/date-and-time-functions.html#function_localtimestamp), and [LOCALTIMESTAMP()](https://dev.mysql.com/doc/refman/8.4/en/date-and-time-functions.html#function_localtimestamp).

Reply:
This explains why the test failed: https://buildkite.com/rails/rails/builds/114408#01939674-b788-4353-8946-8315efd2d644
![](https://screenshot.click/05-18-z3rxc-nh0zv.png)

Reply:
`now()` works for both MySQL 8 and MariaDB. Problem with MariaDB is that it will accept `now` but then return `current_timestamp` when you fetch the column definition from the DB.